tests/integration/test_interface.py

```diff
@@ -300,6 +302,164 @@ def test_dataset_filter_by_year_with_full_dates(mocker, dataset_name: str):
+    if "TODO" in dataset_name:
+        pytest.skip(reason=dataset_name)
+    dataset = DATASETS.get_dataset(dataset_name)
+    data = request.getfixturevalue(f"sample_data_{dataset_name}_state_edit")
+    _generate_non_sample_data_root(dataset_name, tmpdir, data)
+
+    # Skip noising (noising can incorrect select another state)
+    mocker.patch("pseudopeople.interface.noise_dataset", side_effect=_mock_noise_dataset)
+
+    noising_function = DATASET_GENERATION_FUNCS[dataset_name]
+    noised_data = noising_function(source=tmpdir)
+
+    assert (noised_data[dataset.state_column] == STATE).any()
+    assert (noised_data[dataset.state_column] != STATE).any()
+
+
+@pytest.mark.parametrize(
+    "dataset_name",
+    [DATASETS.census.name, DATASETS.tax_w2_1099.name, DATASETS.wic.name, "TODO: tax_1040"],
+)
+def test_dataset_filter_by_state_and_year(mocker, tmpdir, request, dataset_name: str):
+    """Test that dataset generation works with state and year filters in conjunction"""
+    if "TODO" in dataset_name:
+        pytest.skip(reason=dataset_name)
+    year = 2030  # not default 2020
+    data = request.getfixturevalue(f"sample_data_{dataset_name}_state_edit")
+    _generate_non_sample_data_root(dataset_name, tmpdir, data)
+    mocker.patch("pseudopeople.interface._extract_columns", side_effect=_mock_extract_columns)
+    mocker.patch("pseudopeople.interface.noise_dataset", side_effect=_mock_noise_dataset)
+    noising_function = DATASET_GENERATION_FUNCS[dataset_name]
+    noised_data = noising_function(source=tmpdir, year=year, state=STATE)
+    dataset = DATASETS.get_dataset(dataset_name)
+    assert (noised_data[dataset.date_column] == year).all()
+    assert (noised_data[dataset.state_column] == STATE).all()
+
+
+@pytest.mark.parametrize(
+    "dataset_name",
+    [DATASETS.acs.name, DATASETS.cps.name],
+)
+def test_dataset_filter_by_state_and_year_with_full_dates(
+    mocker, tmpdir, request, dataset_name: str
+):
+    """Test that dataset generation works with state and year filters in conjunction"""
+    year = 2030  # not default 2020
+    data = request.getfixturevalue(f"sample_data_{dataset_name}_state_edit")
+    _generate_non_sample_data_root(dataset_name, tmpdir, data)
+    mocker.patch("pseudopeople.interface._extract_columns", side_effect=_mock_extract_columns)
+    mocker.patch("pseudopeople.interface.noise_dataset", side_effect=_mock_noise_dataset)
+    noising_function = DATASET_GENERATION_FUNCS[dataset_name]
+    noised_data = noising_function(source=tmpdir, year=year, state=STATE)
+    dataset = DATASETS.get_dataset(dataset_name)
+    date_format = COLUMNS.get_column(dataset.date_column).additional_attributes.get(
+        "date_format"
+    )
+    if date_format:
+        # The date is a string type, and we cannot expect to use datetime objects
+        # due to the month/day swaps
+        years = noised_data[dataset.date_column].str[0:4].astype(int)
```

Comment:
Note this is potentially brittle if we have some noised date column with a different format. Maybe add a note mentioning that to guide a future engineer if this test unexpectedly breaks. 

I like the comment you already have though, it made it easier to understand the test!